### PR TITLE
Make HttpPipelineEventSource internal, add initial tests 

### PR DESCRIPTION
**Pull request by pakrym**
Initial steps in direction of https://github.com/Azure/azure-sdk-for-net/issues/5489

1. Make EventSource private
2. Remove it from the base type
3. Remove unused members
6. Add initial slightly more detailed tests

cc @KrzysztofCwalina

### CODE REVIEW

**Comment by KrzysztofCwalina** [comment]
I think removal of the LoggingPolicy is an architectural decision we should first discuss at the architecture meeting. The set of policies is something that we wanted to keep consistent between languages.

File: src/SDKs/Azure.Base/data-plane/Azure.Base/Azure.Base.csproj [hunk 1]
```diff
@@ -26,4 +26,8 @@
     <PackageReference Include="System.Numerics.Vectors" Version="4.5.0" />
     <PackageReference Include="System.Runtime.CompilerServices.Unsafe" Version="4.5.2" />
   </ItemGroup>
+
+  <ItemGroup>
+    <InternalsVisibleTo Include="Azure.Base.Tests" />
```
Comment (KrzysztofCwalina) [inline src/SDKs/Azure.Base/data-plane/Azure.Base/Azure.Base.csproj]: Why do we need this?

Comment (pakrym) [reply src/SDKs/Azure.Base/data-plane/Azure.Base/Azure.Base.csproj]: HttpPipelineEventSource is now internal and used in test.

Comment (pakrym) [reply src/SDKs/Azure.Base/data-plane/Azure.Base/Azure.Base.csproj]: Oh wait, my bad. That's my failed attempt at puitting attribute into csproj. Will revert.

File: src/SDKs/Azure.Base/data-plane/Azure.Base/Http/Pipeline/HttpClientTransport.cs [hunk 2]
```diff
@@ -32,10 +35,26 @@ public sealed override async Task ProcessAsync(HttpMessage message)
...
+            if (!responseMessage.IsSuccessStatusCode) {
+                s_eventSource.ErrorResponse(message);
+            }
+
+            var elapsedMilliseconds = (after - before) * 1000 / Stopwatch.Frequency;
+            if (elapsedMilliseconds > s_delayWarningThreshold) {
+                s_eventSource.ResponseDelay(message, elapsedMilliseconds);
+            }
+
+            s_eventSource.ProcessingResponse(message);
```
Comment (KrzysztofCwalina) [inline src/SDKs/Azure.Base/data-plane/Azure.Base/Http/Pipeline/HttpClientTransport.cs]: As a result of removing the logging policy, the networking code and the logging code are now all intermixed. It's not a ship stopper, but it is a small negative IMO.

Comment (pakrym) [reply src/SDKs/Azure.Base/data-plane/Azure.Base/Http/Pipeline/HttpClientTransport.cs]: An interesting tangent from this is that we might want to log the message both when entering and exiting (being sent by transport) the pipeline.

Comment (KrzysztofCwalina) [reply src/SDKs/Azure.Base/data-plane/Azure.Base/Http/Pipeline/HttpClientTransport.cs]: I agree

**Comment by pakrym** [comment]
@KrzysztofCwalina :up::date:

**Comment by KrzysztofCwalina** [comment]
I thought we talked about bringing back the LoggingPolicy property on options. I still see it removed.

**Comment by pakrym** [comment]
I brought it back in the last commit yesterday

**Review by KrzysztofCwalina — APPROVED** [review]

**Review by bterlson — APPROVED** [review]

**Pull request merged**